Userland/Libraries/LibCore/Stream.h

```diff
@@ -671,11 +671,11 @@ class BufferedHelper {
         if (!stream().is_open())
             return Error::from_errno(ENOTCONN);
 
-        if (buffer.is_empty())
-            return Error::from_errno(ENOBUFS);
-
         auto const candidate = TRY(find_and_populate_until_any_of(candidates, buffer.size()));
 
+        if (buffer.is_empty())
+            return buffer;
+
```

Comment:
Considering that `find_and_populate_until_any_of` with `max_offset` being 0 basically would be a no-op anyways, I think it would be more logical to keep the empty buffer check where it is and just change the response (as with the other check).

```cpp
[...]

if (buffer.is_empty())
    return buffer;

auto const candidate = TRY(find_and_populate_until_any_of(candidates, buffer.size()));

[...]
```